src/test/trackexport_test.cpp

```diff
@@ -0,0 +1,283 @@
+// Exercises the track export worker.
+// Just uses temp directories rather than trying to mock the filesystem.
+
+#include "test/trackexport_test.h"
+
+#include <QDebug>
+#include <QScopedPointer>
+#include <QTemporaryFile>
+
+FakeOverwriteAnswerer::~FakeOverwriteAnswerer() { }
+
+void FakeOverwriteAnswerer::slotProgress(QString filename, int progress, int count) {
+    m_progress_filename = filename;
+    m_progress = progress;
+    m_progress_count = count;
+}
+
+void FakeOverwriteAnswerer::slotAskOverwriteMode(
+            QString filename,
+            std::promise<TrackExportWorker::OverwriteAnswer>* promise) {
+    auto it = m_answers.find(filename);
+    // Make sure this filename is in the map, and then remove it.
+    // The ASSERT_EQ macro has trouble with this comparison.
+    ASSERT_FALSE(it == m_answers.end());
+    TrackExportWorker::OverwriteAnswer answer = m_answers[filename];
```

Comment:
Since this is a test no need to fix -- just pointing out that this could be *it and remove could be m_answers.erase(it).